src/snowflake/snowpark/session.py

```diff
@@ -1158,9 +1159,9 @@ def get_req_identifiers_list(
 
                 # Add to packages dictionary
                 if name in result_dict:
-                    if result_dict[name] != str(package):
+                    if version is not None and result_dict[name] != str(package):
```

Comment:
Is it possible to add the same package + version when it already exists? Or do we already deduplicate the packages (e.g. via a dict)?

Reply:
we first add packages to `result_dict` that are available in snowflake and are added via `add_packages()` (or other such methods). Then, we pip install other packages and figure out dependencies. Now there can be dependency clashes if requested dependency add via add_package is `package_a==x.y.1` and inferred dependency via pip installation is `package_a==x.y.2`. In this case, resolve packages should throw an error.

In the updated logic, we ignore case when inferred dependency is not version specific, in other words inferred depndency `package_a` is compatible with `package_a==x.y.1`. This case would throw error in resolve packages earlier but passes now.

Reply:
If we have a `package_a==x.y.1` from `add_package` and the same `package_a==x.y.1` from `pip`, we should pass but I'm not sure if this is a legit case, or if our code has this scenario covered already (it might be covered by the original `result_dict[name] != str(package)`).

Reply:
that's right, in the case you mentioned, `result_dict[name] == str(package)`